libraries/botbuilder-dialogs/src/dialogContext.ts

```diff
@@ -148,10 +148,10 @@ export class DialogContext {
      * return await dc.prompt('confirmPrompt', `Are you sure you'd like to quit?`);
      * ```
      * @param dialogId ID of the prompt to start.
-     * @param promptOrOptions Initial prompt to send the user or a set of options to configure the prompt with..
-     * @param choicesOrOptions (Optional) array of choices associated with the prompt.
+     * @param promptOrOptions (Optional) Initial prompt to send the user or a set of options to configure the prompt with..
+     * @param choices (Optional) array of choices associated with the prompt.
      */
-    public async prompt(dialogId: string, promptOrOptions: string|Partial<Activity>|PromptOptions): Promise<DialogTurnResult>;
+    public async prompt(dialogId: string, promptOrOptions?: string|Partial<Activity>|PromptOptions, choices?: (string|Choice)[]): Promise<DialogTurnResult>;
```

Comment:
Why is `promptOrOptions` now optional? The function marks it as [required](https://github.com/Microsoft/botbuilder-js/pull/721/files#diff-957a2549a16757f7986d71b20d222995R157). 

Additionally, this results in one typing, where choices is optional. Can you keep the old typing and add an overload that includes choices?

> I expected to see multiple choices for prompt() with different sets of parameters. Or at least an optional choices parameter.
> - [navzam](https://github.com/Microsoft/botbuilder-js/issues/697#issue-390784742)

If you need an example, please see PR #647.